src/config.rs

```diff
@@ -590,12 +617,12 @@ impl Cfg {
         Ok(None)
     }
 
-    fn parse_override_file<S: AsRef<str>>(contents: S) -> Result<OverrideFile> {
+    fn parse_override_file<S: AsRef<str>>(contents: S, toml_only: bool) -> Result<OverrideFile> {
         let contents = contents.as_ref();
 
         match contents.lines().count() {
             0 => Err(ErrorKind::EmptyOverrideFile.into()),
-            1 => {
+            1 if !toml_only => {
```

Comment:
With the enum rather than the boolean, this would likely be nicely done as:

```rust
match (contents.lines.count(), parse_mode) {
    (0, _) => ...
    (1, ParseMode::Both) => {...}
    _ => {...}
}
```